subprojects/language-native/src/main/java/org/gradle/language/swift/plugins/SwiftLibraryPlugin.java

```diff
@@ -141,23 +144,25 @@ public String call() throws Exception {
                             runtimeAttributes.attribute(DEBUGGABLE_ATTRIBUTE, buildType.isDebuggable());
                             runtimeAttributes.attribute(OPTIMIZED_ATTRIBUTE, buildType.isOptimized());
                             runtimeAttributes.attribute(LINKAGE_ATTRIBUTE, linkage);
-                            runtimeAttributes.attribute(OperatingSystemFamily.OPERATING_SYSTEM_ATTRIBUTE, operatingSystem);
+                            runtimeAttributes.attribute(OperatingSystemFamily.OPERATING_SYSTEM_ATTRIBUTE, targetMachine.getOperatingSystemFamily());
+                            runtimeAttributes.attribute(MachineArchitecture.ARCHITECTURE_ATTRIBUTE, targetMachine.getArchitecture());
 
                             AttributeContainer linkAttributes = attributesFactory.mutable();
                             linkAttributes.attribute(Usage.USAGE_ATTRIBUTE, linkUsage);
                             linkAttributes.attribute(DEBUGGABLE_ATTRIBUTE, buildType.isDebuggable());
                             linkAttributes.attribute(OPTIMIZED_ATTRIBUTE, buildType.isOptimized());
                             linkAttributes.attribute(LINKAGE_ATTRIBUTE, linkage);
-                            linkAttributes.attribute(OperatingSystemFamily.OPERATING_SYSTEM_ATTRIBUTE, operatingSystem);
+                            linkAttributes.attribute(OperatingSystemFamily.OPERATING_SYSTEM_ATTRIBUTE, targetMachine.getOperatingSystemFamily());
+                            linkAttributes.attribute(MachineArchitecture.ARCHITECTURE_ATTRIBUTE, targetMachine.getArchitecture());
 
-                            NativeVariantIdentity variantIdentity = new NativeVariantIdentity(variantName, library.getModule(), group, version, buildType.isDebuggable(), buildType.isOptimized(), operatingSystem,
+                            NativeVariantIdentity variantIdentity = new NativeVariantIdentity(variantName, library.getModule(), group, version, buildType.isDebuggable(), buildType.isOptimized(), targetMachine,
                                 new DefaultUsageContext(variantName + "Link", linkUsage, linkAttributes),
                                 new DefaultUsageContext(variantName + "Runtime", runtimeUsage, runtimeAttributes));
                             // TODO: publish Swift libraries
                             // library.getMainPublication().addVariant(variantIdentity);
 
-                            if (DefaultNativePlatform.getCurrentOperatingSystem().toFamilyName().equals(operatingSystem.getName())) {
-                                ToolChainSelector.Result<SwiftPlatform> result = toolChainSelector.select(SwiftPlatform.class);
+                            if (DefaultNativePlatform.getCurrentOperatingSystem().toFamilyName().equals(targetMachine.getOperatingSystemFamily().getName())) {
+                                ToolChainSelector.Result<SwiftPlatform> result = toolChainSelector.select(SwiftPlatform.class, targetMachine);
```

Comment:
This made me think... what happens if you try to target 32-bit macOS with Swift? Do we assume the Swift tool chain can do 32-bit?

Reply:
Good point.  Yeah, the swift toolchain wasn't doing any checks on target architecture, so it would just accept anything.  Looking around, there doesn't really seem to be any support for cross-platform compiling, so for now, I just changed this to fail if the target platform is not the current platform.  We can follow this up with some sort of compiler probe if the cross-platform case emerges.

Reply:
Should we think about having a read-only architecture field on the target platform for Swift binary? It may just add confusion to the mix if we allow modifying the architecture. It will probably be useful when we target iOS but until then, it serves no purpose to allow modification. We could finalize the value right away.

Reply:
Per our conversation, IIUC, you're suggesting that we add the current os/arch target machine and then immediately finalize the target machines provider.  I don't think we can do this because we need to allow multiple target machines for the purposes of publishing (e.g. targets for both linux and macOS).  So even though we don't build for the other target machine, we still need to be able to declare it so we can publish the metadata.